admin_guide/backup_restore.adoc

```diff
@@ -0,0 +1,115 @@
+//
+// In the following text, there are questions of the form:
+//   //??? QUESTION
+// Please feel free to make a line-comment to answer them, in addition
+// to any other (line-)comments on the correctness of the text.
+//
+// - Usually, the question pertains to the text preceding it.
+//   Questions pertaining to following text are explicitly noted.
+//
+// - There are bunch of questions at the end.
+//
+// Thanks for your cooperation on this (experimental) method of
+// refining the documentation.  Hopefully it will bear good fruit.
+
+== Overview
+
+In {product-title}, you can
+do _backup_, saving state to separate storage,
+and _restore_, recreating state from separate storage,
+at the cluster level.
+The full state of a cluster installation includes:
+- etcd data on each master
+- API objects
+//??? Anything else?
+//??? What about (non-master) nodes?
+
+[NOTE]
+Project-level backup/restore will be available in a future release.
+//??? Is this true?
+
+This section does not cover backup and restore of
+link:../install_config/persistent_storage/index.html[persistent storage].
+//??? But should it?
+
+
+[[cluster-backup]]
+== Cluster Backup
+
+//??? (for this section) Is the ordering (etcd, then API objects) correct?
+
+. Create an *etcd* backup on each master:
++
+----
+# yum install etcd
+# etcdctl backup \
+    --data-dir /var/lib/origin/openshift.local.etcd \
+    --backup-dir /var/lib/origin/openshift.local.etcd.bak
+----
+
+. Create a template for all cluster API objects:
++
+====
+----
+$ oc export all \
+    --exact \//<1>
+    --all-namespaces \
+    --as-template=cluster.template
+----
+<1> Preserve fields that may be cluster specific,
+such as service `portalIP`s or generated names.
+====
+//??? pkg/cmd/cli/cmd/export.go line 76 says:
+//    cmd.Flags().Bool("all", true, "DEPRECATED: all is ignored, specifying a resource without a name selects all the instances of that resource")
+//    What does "deprecated" mean for the user?  (Can ‘all’ be used, anyway?)
+
+
+[[cluster-restore]]
+== Cluster Restore
+
+//??? (for this section) Is the ordering (API objects, then etcd) correct?
+
+. Reinstall {product-title}.
+//??? Is there a better way to "zero out" the cluster?
+This should be done in the
+link:../install_config/install/index.html[same way]
+that {product-title} was previously installed.
+
+. Create the API objects for the cluster:
++
+----
+$ oc process -f cluster.template
+----
+//??? Other flags?
+
+. Restore from the *etcd* backup on each master:
++
+----
+# etcdctl restore \
+    --backup-dir /var/lib/origin/openshift.local.etcd.bak \
+    --data-dir /var/lib/origin/openshift.local.etcd
+----
+//??? This is a guess based on <https://lwn.net/Articles/631630/> (!).
+//    What am i missing?
+
+
+// ---------------------------------------------------------------------
+//??? Does the cluster need to be "quiescent" for backup/restore/both?
+//??? Generally, what are the required conditions for a successful backup/restore?
```

Comment:
There needs to be a test plan for this, before we publish.
On May 26, 2016 6:52 PM, "Thien-Thi Nguyen" <notifications@github.com>
wrote:

> In admin_guide/backup_restore.adoc
> <https://github.com/openshift/openshift-docs/pull/2140#discussion_r64834607>
> :
>
> > +//??? Other flags?
> > +
> > +. Restore from the *etcd* backup on each master:
> > ++
> > +----
> > +# etcdctl restore \
> > +    --backup-dir /var/lib/origin/openshift.local.etcd.bak \
> > +    --data-dir /var/lib/origin/openshift.local.etcd
> > +----
> > +//??? This is a guess based on <https://lwn.net/Articles/631630/> (!).
> > +//    What am i missing?
> > +
> > +
> > +// ---------------------------------------------------------------------
> > +//??? Does the cluster need to be "quiescent" for backup/restore/both?
> > +//??? Generally, what are the required conditions for a successful backup/restore?
>
> @sferich888 <https://github.com/sferich888>
> The QA contact on the BZ was originally @vikram-redhat
> <https://github.com/vikram-redhat> and now @xltian
> <https://github.com/xltian> -- however, because this is a sort of
> "non-feature", i'm not sure if there is/was a testing plan.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/2140/files/503f520bb05a7700f9587f3835174762f0b9c43c#r64834607>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABUZmgl_ZwE4gvkndwlJRMFOgBFAayNVks5qFiQggaJpZM4IjZF_>
> .
>